### DISCUSSION TITLE
Filter by (extra) type

### DISCUSSION THREAD

**Discussion by denilsonsa** [Kalanyr/gogrepo#101] [Ideas]
I wanted to download all my soundtracks from GOG, but none of the game files and none of the wallpapers and none of the art books…

I know I can skip the game files:

    python gogrepoc.py download -skipgalaxy -skipstandalone -skipshared -skipgames

But there is no parameter to filter out specific kinds of extras. Yes, there is a file filter, but it doesn't work, as some soundtracks have `_ost` in the filename while others have `_soundtrack`.

As a quick hack, I've added this code [around line 1950](https://github.com/Kalanyr/gogrepoc/blob/62a141892ea138810578e368bc2a5ec4e17b9b75/gogrepoc.py#L1948):

```python
        item.extras = [ x for x in item.extras if 'soundtrack' in x.desc.lower() ]
```

If this idea gets implemented, we could have a new command to list all the possible values for the `extras[i].desc`, and then have a new parameter to the `download` command to filter by the specified values.

**Comment by Kalanyr** [comment]
Interesting idea, I'll do a check through my manifest when I get a chance
and verify that desc is well-formed enough for this to work consistently.

**Comment by denilsonsa** [comment]
I'm looking at the manifest now. Unfortunately, it's not well-formatted enough. It's still possible to get some patterns for most soundtracks (like `re.search(r'soundtrack|music', ...)`), as well as for avatars, wallpapers, manuals and a few other common extras.
